app/api/webcal_api.rb

```diff
@@ -81,7 +81,8 @@ class WebcalApi < Grape::API
             projects: { user_id: webcal.user_id },
             units: { active: true }
           )
-          .where('? BETWEEN units.start_date AND units.end_date', Time.zone.now)
+          .where('tasks.project_id is null or tasks.project_id = projects.id')   # eager_load(:tasks) only tasks of :projects
+          .where('? BETWEEN units.start_date AND units.end_date', Time.zone.now) # Current units
```

Comment:
This is equivalent to the `Unit.current` scope.

https://github.com/OnTrack-UG-Squad/doubtfire-api/blob/development/app/models/unit.rb#L148

Is it possible to somehow include `Unit.current` in the join so as to not redefine the scope here?

Reply:
Not sure this would be beneficial. In the end you are just creating a query here to report data.